src/main.rs

```diff
@@ -246,5 +312,40 @@ fn main() {
     let r = afd_poll(afd_helper_handle, &mut poll_info, &mut overlapped);
     println!("{:?}", r);
 
-    //GetQueuedCompletionStatusEx
+    let mut completion_count: DWORD = 0;
+    let mut iocp_events: [OVERLAPPED_ENTRY; 256] = [OVERLAPPED_ENTRY::default(); 256];
+    let r = unsafe {
+        GetQueuedCompletionStatusEx(
+            iocp,
+            iocp_events.as_mut_ptr(),
+            iocp_events.len() as ULONG,
+            &mut completion_count as *mut _,
+            INFINITE,
+            FALSE,
+        )
+    };
+
+    println!("Return value: {:?}", r);
+    println!("completion_count: {:?}", completion_count);
+    println!("iocp_events: ");
+    for ele in iocp_events[0..completion_count as usize].iter() {
+        println!("  Event: ");
+        println!("    lpCompletionKey: {:?}", ele.lpCompletionKey);
+        println!("    lpOverlapped: {:?}", ele.lpOverlapped);
+        /* ignore it for now
+        if NULL as _ != ele.lpOverlapped {
+            println!("      *lpOverlapped: {:?}", *ele.lpOverlapped);
+        }
+        */
+        println!("    Internal: {:?}", ele.Internal);
+        println!(
+            "    dwNumberOfBytesTransferred: {:?}",
+            ele.dwNumberOfBytesTransferred
+        );
+    }
+
+    //WSACleanup is used to terminate the use of the WS2_32 DLL.
+    unsafe {
+        WSACleanup();
```

Comment:
This is only (minimally) relevant when building a shared library, not when building an executable.
Also, nobody ever unloads winsock, just like nobody ever "unloads" libc. Don't waste your time with it.

Reply:
Okay, I will remove that.